### PR TITLE
Oauth proxy

### PR DESCRIPTION
**Pull request by ShovelHand**
Added OAuth-Proxy sidecar to project.

### CODE REVIEW

File: template/appassessment-template.yaml [hunk 1]
```diff
@@ -47,6 +47,9 @@ objects:
     name: appassessment-pvc
     labels:
       app: appassessment-app
+    name: appassessment-cert-pvc
+    labels:
+      app: appassessment-app
```
Comment (StevenBarre) [inline template/appassessment-template.yaml:52]: A PVC can't have two names and labels

File: template/appassessment-template.yaml [hunk 2]
```diff
@@ -103,6 +110,28 @@ objects:
...
+            args:
+              - --https-address=:8888
+              - --provider=openshift
+              - --openshift-service-account=proxy
+              - --upstream=http://localhost:8080
+              - --tls-cert=/etc/tls/private/tls.crt
+              - --tls-key=/etc/tls/private/tls.key
+              - --cookie-secret=SECRET
+              - --openshift-sar={"resource":"services","verb":"get"}
+            image: quay.io/openshift/origin-oauth-proxy:4.1
```
Comment (StevenBarre) [inline template/appassessment-template.yaml:123]:
```suggestion
            image: quay.io/openshift/origin-oauth-proxy:4.7
```
Use a newer version.

File: template/appassessment-template.yaml [hunk 3]
```diff
@@ -103,6 +110,28 @@ objects:
...
+            imagePullPolicy: IfNotPresent
+            volumeMounts:
+              - name: appassessment-tls
+                mountPath: /etc/tls/private
+            volumes:
+            - secret:
+              name: appassessment-tls
```
Comment (StevenBarre) [inline template/appassessment-template.yaml:130]: Duplicate of L89

File: build/buildconfig.yaml [hunk 4]
```diff
@@ -30,7 +30,7 @@ spec:
     type: Git
     git:
       uri: 'https://github.com/bcgov/appassessment'
-      ref: template-improvements
+      ref: oauth-proxy
```
Comment (StevenBarre) [inline build/buildconfig.yaml:33]: Should probably point to the `main` branch after this is merged.

**Review by StevenBarre — CHANGES_REQUESTED** [review]

File: template/appassessment-template.yaml [hunk 5]
```diff
@@ -103,6 +110,28 @@ objects:
...
+          - name: oauth-proxy 
+            args:
+              - --https-address=:8888
+              - --provider=openshift
+              - --openshift-service-account=proxy
+              - --upstream=http://localhost:8080
+              - --tls-cert=/etc/tls/private/tls.crt
+              - --tls-key=/etc/tls/private/tls.key
+              - --cookie-secret=SECRET
+              - --openshift-sar={"resource":"services","verb":"get"}
```
Comment (StevenBarre) [inline template/appassessment-template.yaml:122]:
```suggestion
              - --openshift-sar={"namespace":"${LICENSE_PLATE}-tools","resource":"services","verb":"get"}
```

File: template/appassessment-template.yaml (hunk 1 above)
Comment (ShovelHand) [reply template/appassessment-template.yaml:52]: removed these lines

File: template/appassessment-template.yaml (hunk 3 above)
Comment (ShovelHand) [reply template/appassessment-template.yaml:130]: removef

**Pull request merged**